### ISSUE TITLE
Duplicate transaction reference error

### ISSUE THREAD

**Issue by devbee001** [Princewil/flutter_paystack_plus#15]
Getting a response code of 400 and a body message of duplicate transaction reference , on any transaction , i verify that i am using a unique reference. 
it happens immediately the text form is clicked using a life secret key - android .

**Comment by Princewil** [comment]
Can you share a screenshot of your code?

**Comment by devbee001** [comment]
sure this is my code - 

  intiatePaystackPayment(
    context,
  ) {
    final uniqueTransRef = const Uuid().v4();

    PayWithPayStack().now(
        context: context,
        secretKey: "sk_live_**************",
        customerEmail: "ayenibolaji07@gmail.com",
        reference: uniqueTransRef,
        callbackUrl: "setup in your paystack dashboard",
        currency: "NGN",
        paymentChannel: ["card"],
        amount: '20000',
        transactionCompleted: () {
          print("Transaction Successful");
        },
        transactionNotCompleted: () {
          print("Transaction Not Successful!");
        });
  }
}

**Comment by Princewil** [comment]
Hmm this is quite unprecedented🤔

Can you try using `DateTime.now().millsecondsSinceEpoch` instead of the Uuid package

**Comment by devbee001** [comment]
Yes I firstly did that, i got the same error message, so i turned to the
Uuid package.

Did you try it to see if you can recreate the problem ?

**Comment by Princewil** [comment]
No I haven't. Maybe I should
I will give you a feedback when I do

Meanwhile, you can clone the example code, link on the package docs. You don't have to change anything, just input your secret and public key and then run the code, let's see if you still have the issue.

**Comment by peterchibunna** [comment]
This only happens when you use a live secret key, everything works as expected when you use a test key. Can you look into it again?

**Comment by Princewil** [comment]
Alright, I'll check it out.

**Comment by peterchibunna** [comment]
More context: Trying to interact with the screen triggers a refresh of the page

**Comment by peterchibunna** [comment]
https://github.com/user-attachments/assets/71f38dc3-da93-45ac-9fd7-06f102b20b6a

**Comment by Princewil** [comment]
Got it!

**Comment by peterchibunna** [comment]
Hi, it's a webview issue. There's a bug with the flutter webview that crashes it when the user tries to type into an input field or when an input field appears on the webview. I've tried to use the InAppWebview widget and it works well.

**Comment by Princewil** [comment]
Glad to hear it!
I may have to change the web view package I used then

**Comment by Princewil** [comment]
You can request PR if you fixed it

**Comment by peterchibunna** [comment]
The implementation of webview widget and its controller is slightly different from that of the inappwebview. I implemented my solution outside of any package. It's like manually initiating a transaction with the reference, amount, email, etc, then open a fullscreen modal when the authorization url becomes available and then load it into the inappwebview, then start monitoring the url changes.

**Comment by Princewil** [comment]
Okay?
So what’s the point?
